modules/gapi/include/opencv2/gapi/render.hpp

```diff
@@ -0,0 +1,54 @@
+#ifndef OPENCV_GAPI_RENDER_HPP
+#define OPENCV_GAPI_RENDER_HPP
+
+#include <string>
+#include <vector>
+
+#include <opencv2/gapi/opencv_includes.hpp>
+#include <opencv2/gapi/util/variant.hpp>
+#include <opencv2/gapi/own/exports.hpp>
+#include <opencv2/gapi/own/scalar.hpp>
+
+namespace cv
+{
+namespace gapi
+{
+
+struct TextEvent
+{
+    std::string text;
+    int x;
+    int y;
+    int ff;
+    double fs;
+    cv::gapi::own::Scalar color;
+    int thick;
+    int lt;
+    bool bottom_left_origin;
+};
+
+struct RectEvent
+{
+    int x;
+    int y;
+    int widht;
+    int height;
+    cv::gapi::own::Scalar color;
+    int thick;
+    int lt;
+    int shift;
+};
+
+using DrawEvent  = util::variant<TextEvent, RectEvent>;
+using DrawEvents = std::vector<DrawEvent>;
+
+GAPI_EXPORTS void render(cv::Mat& bgrx, const std::vector<DrawEvent>& events);
+
+// FIXME Specify the signature for NV12 case
+GAPI_EXPORTS void render(cv::Mat& y_plane, cv::Mat& uv_plane , const std::vector<DrawEvent>& events);
+
+} // namespace gapi
+} // namespace cv
+
+
+#endif // OPENCV_GAPI_RENDER_HPP
```

Comment:
I would expect more comments in this file.

Reply:
+1. makes sense to document fields of our DrawEvents in the way how those are documented in the original OpenCV counterparts.

Reply:
Fixed

Reply:
There are warnings in Doxygen now: https://pullrequest.opencv.org/buildbot/builders/precommit_docs/builds/20842

Also, `@param` is a way to document function arguments, not structure fields.

This is how fields are documented in doxygen: https://wiki.scilab.org/Doxygen%20documentation%20Examples

And this is how it looks now in the generated documentation (see above build):

![image](https://user-images.githubusercontent.com/144187/59441795-ca339380-8e01-11e9-97fb-f2e90cb4663f.png)